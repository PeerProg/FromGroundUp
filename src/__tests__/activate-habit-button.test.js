import React from 'react';
import { render, cleanup } from 'react-testing-library';
import ActivateHabitButton from '../components/activate-habit-button';

afterEach(cleanup);

test('should render the activate habit button correctly', () => {
  const { container } = render(<ActivateHabitButton />);
  expect(container).toMatchSnapshot();
});

test('should render the activate habit button with correct text', () => {
  const { rerender, getByTestId } = render(
    <ActivateHabitButton habitActive={true} />
  );

  const buttonNode = getByTestId('activate-habit-button');

  expect(buttonNode.innerHTML).toEqual('Stop');

  rerender(<ActivateHabitButton habitActive={false} />);
  expect(buttonNode.innerHTML).toEqual('Start');
});
