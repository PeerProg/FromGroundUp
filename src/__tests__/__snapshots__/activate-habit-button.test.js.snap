// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the activate habit button correctly 1`] = `
<div>
  <button
    type="button"
    class="btn btn-secondary"
    data-testid="activate-habit-button"
  >
    Start
  </button>
</div>
`;
